db/migrate/20210609114653_add_cached_counts.rb

```diff
@@ -0,0 +1,17 @@
+class AddCachedCounts < ActiveRecord::Migration[5.2]
+  def up
+    add_column :policies, :total_host_count, :integer
+    add_column :policies, :test_result_host_count, :integer
+    add_column :policies, :compliant_host_count, :integer
+    add_column :policies, :unsupported_host_count, :integer
+
+    Policy.find_each(&:update_counters!)
+  end
```

Comment:
Did I make some comment earlier about these being `null: false`? I can't remember if it was this PR or another one, but don't you think it makes more sense to have a default value of 0 for these counts and force them to be non-null? I think my previous comment was about score, which doesn't make as much sense as 0 when there's no data.

Reply:
The other one, but I can definitely put the `where` here.

Reply:
Well, I mean, don't you think this migration makes sense to add `null: false, default: 0` or similar? Host counts being 0 by default make more sense than score being 0 by default.

Reply:
True

Reply:
Still an issue as the `policy` can be `nil` and then these fields ale `nil` as well and GQL doesn't like it, even if I do `allow_nil` in the delegation. So yeah, I do need the `to_i` but I will keep the `default: 0` as I like this :+1: